Adding mutation and query to configurable and downloadable product endpoints (#3867)
magento/devdocs: guides/v2.3/graphql/reference/configurable-product.md

title: ConfigurableProduct endpoint
---

> The `ConfigurableProduct` endpoint defines a query and mutation for configurable products.

Reviewer: ```suggestion
The `ConfigurableProduct` endpoint defines a mutation for adding configurable products to a cart. It also extends the `ProductInterface` so that attributes that are specific to configurable produ sects can be queried in a `products`
```

title: ConfigurableProduct endpoint
---

The `ConfigurableProduct` endpoint defines a mutation for adding configurable products to a cart. It also extends the `ProductInterface` so that attributes that are specific to configurable products can be queried in a `products` search.

## Query
The `products` query returns configurable product-specific information when you perform a `products` search.

### Syntax

> Add the following inline fragment to the output section of your products query to return information specific to configurable products:

Reviewer: ```suggestion
Add the following inline fragment to the output section of your `products` query to return information specific to configurable products:
```

`label` | String | A string that describes the configurable product option. It is displayed on the UI.
`position` | Int | A number that indicates the order in which the attribute is displayed
`product_id` | Int | This is the same as a product's 'id' field

> `values` | [[ConfigurableProductOptionsValues]](#configProdOptionsValues) | An array that defines the value_index codes assigned to the configurable product

Reviewer: ```suggestion
`values` | [[ConfigurableProductOptionsValues]](#configProdOptionsValues) | An array that defines the `value_index` codes assigned to the configurable product
```

          product {
            id

>             categories {

Reviewer: The categories object produces redundant output. Let's save a few electrons by removing that object from the query.

I can't comment on text that isn't near changed text, but also delete store_label, default_label, and use_default_value (lines 88-90).
